lib/install/management/appliance.go

```diff
@@ -611,6 +668,12 @@ func (d *Dispatcher) makeSureApplianceRuns(conf *config.VirtualContainerHostConf
 	if !ip.IsUnspecifiedIP(conf.ExecutorConfig.Networks["client"].Assigned.IP) {
 		d.HostIP = conf.ExecutorConfig.Networks["client"].Assigned.IP.String()
 		log.Debug("Obtained IP address for client interface: %q", d.HostIP)
+
+		// wait till the appliance is fully initialized
+		initSuccess := waitForApplianceInit(d.HostIP, d.DockerPort)
```

Comment:
There are other codes to retrieve components status. It's fine to keep current logic, but need to move your code to the end of the method.
And return error if service is unreachable.